pkg/autoscaler/aggregation/bucketing.go

```diff
@@ -60,22 +60,32 @@ func (t *TimedFloat64Buckets) IsEmpty() bool {
 	return len(t.buckets) == 0
 }
 
-// GetAndLock returns the buckets and secures them via a mutex. The contents of
-// the returned map can be modified and the modifications will be visible to other
-// readers. Unlock must be called after the client is done reading/manipulating
-// the data.
-func (t *TimedFloat64Buckets) GetAndLock() map[time.Time]Float64Bucket {
-	t.bucketsMutex.Lock()
-	return t.buckets
+// Process processes all buckets saved currently with the given Accumulutor functions.
```

Comment:
Nit: s/Accumulutor/Accumulator/

Id reword this comment a bit too since Process is a pretty undescriptive name - maybe 'calls each accumulator once for each bucket'

Reply:
Good one, I renamed it to `ForEachBucket` as well for posterity.